src/Wallabag/CoreBundle/Controller/EntryController.php

```diff
@@ -20,6 +20,48 @@
 
 class EntryController extends Controller
 {
+    /**
+     * @Route("/mass", name="mass_action")
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function massAction(Request $request)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $values = $request->request->all();
+
+        $action = 'toggle-read';
+        if (isset($values['toggle-star'])) {
+            $action = 'toggle-star';
+        } elseif (isset($values['delete'])) {
+            $action = 'delete';
+        }
+
+        if (isset($values['entry-checkbox'])) {
```

Comment:
I would rather move the `$redirectUrl` above that condition and switch the condition check to avoid a big jump in indentation:

```php
$redirectUrl = $this->get('wallabag_core.helper.redirect')->to($request->headers->get('referer'));

if (!isset($values['entry-checkbox'])) {
    return $this->redirect($redirectUrl);
}

foreach ( ... ) {

}

return $this->redirect($redirectUrl);
```